website/docs/core-concepts/connecting-to-data-sources/authentication/connect-to-apis.md

```diff
@@ -4,24 +4,110 @@ description: You can hit any REST endpoint available on the public internet
+
 :::tip
-If your API is not available on the public internet, you must expose it via a service such as [https://ngrok.com/](https://ngrok.com/)
+If you want to access an API that's not publicly available on the internet, you can use a service like [ngrok](https://ngrok.com/) to expose it. For more information, see [How to work with local APIs on Appsmith](/advanced-concepts/more/how-to-work-with-local-apis-on-appsmith).
 :::
 
-## API Editor
+## REST API query editor
 
-The API pane is a REST interface that allows you to create and modify your existing APIs. All [REST HTTP methods](https://www.w3schools.in/restful-web-services/rest-methods) are supported and API values can be configured in the headers, params, and body fields
+The API pane is a user-friendly interface for creating and managing RESTful APIs. It supports all standard [REST HTTP methods](https://www.w3schools.in/restful-web-services/rest-methods) and allows you to configure API values in the headers, query parameters, and request body fields.
 
 ![](</img/create_api_(1).gif>)
 
-## Importing CURL Commands
+## Configure REST API query
+
+Configuring an API query involves specifying the various components that make up the request and the desired response. Here is an overview of the main elements you need to consider when configuring an API query.
+
+### Name
+
+When you create a query in Appsmith, a default name is provided for you. You can change this name by clicking on the pencil icon. It's important to make sure that the name is unique, as it serves as an identifier for queries on the page.
+
+### Method
+
+A method refers to the type of request being made to the REST API endpoint. Appsmith supports all the standard methods, including GET, POST, PUT, DELETE, and PATCH. When setting up an API query in Appsmith, you can select the desired method from a dropdown menu before adding the URL.
+
+It's important to choose the appropriate method based on the task you want to accomplish, as each method has a specific meaning and can behave differently depending on the API you are using.
+
+
+### URL
+
+The URL Path provides a way to access the API and its functions. To use this field, you can enter the API URL that you want to access. For example, if you want to access the Appsmith mock API, you can enter the URL as `https://mock-api.appsmith.com`.
+
+### Headers
+
+HTTP headers are a critical part of the API request and response, as they contain meta-data about the API request and response. Headers carry the following information:
+
+* Request Authorization;
+* Response Caching;
+* Response Cookies.
+
+It's important to be familiar with the major request and response headers, as you may need to set request headers when testing an API, and you may need to set assertions against response headers to ensure that the correct headers are being returned.
```

Comment:
Can we remove this line - `and you may need to set assertions against response headers to ensure that the correct headers are being returned.` - reason being that at this moment, IMO, we don't provide a way to the user to check the response header values.

Reply:
done